src/Annotation/Reader.php

```diff
@@ -0,0 +1,394 @@
+     *
+     * If no annotation is found the method returns null.
+     * Array structure:
+     *  [
+     *      propertyName1 => [
+     *          Annotation1,
+     *          Annotation1,
+     *          [...]
+     *      ],
+     *      propertyName2 => [
+     *          Annotation1,
+     *          Annotation2,
+     *          [...]
+     *      ],
+     *      [...]
+     *  ]
+     *
+     * @return array|null
+     *
+     * @throws AppException
+     */
+    public function getPropertiesAnnotations(): ?array
+    {
+        $reflectionProperties = $this->reflectionClass->getProperties();
+
+        if (empty($reflectionProperties)) {
+            return null;
+        }
+
+        $annotations = [];
+
+        foreach ($reflectionProperties as $reflectionProperty) {
+            $result = $this->annotationManager->getAnnotations(
+                $reflectionProperty->getDocComment()
+            );
+
+            if ($result !== null) {
+                $annotations[$reflectionProperty->getName()] = $result;
+            }
+        }
+
+        return $annotations;
+    }
+
+    /**
+     * Returns an array of $annotation objects from all properties.
+     *
+     * If true is passed to $unique, the method will just return the first occurrence of $annotation
+     * If no annotation is found the method returns null.
+     * Array structure:
+     *  [
+     *      propertyName1 => [
+     *          Annotation1,
+     *          Annotation1,
+     *          [...]
+     *      ],
+     *      propertyName2 => [
+     *          Annotation1,
+     *          Annotation2,
+     *          [...]
+     *      ],
+     *      [...]
+     *  ]
+     *
+     * @param string    $annotation     Annotation tag name
+     * @param bool      $unique         Option.
+     *
+     * @return array|null
+     *
+     * @throws AppException
+     */
+    public function getPropertiesAnnotation(string $annotation, bool $unique = false)
+    {
+        $reflectionProperties = $this->reflectionClass->getProperties();
+
+        if (empty($reflectionProperties)) {
+            return null;
+        }
+
+        return $this->getAnnotations($reflectionProperties, $annotation, $unique);
+    }
+
+    /**
+     * @param array  $reflections
+     * @param string $annotation
+     * @param bool   $unique
+     *
+     * @return array
+     *
+     * @throws AppException
+     */
+    private function getAnnotations(array $reflections, string $annotation, bool $unique)
+    {
+        $annotations = [];
+
+        foreach ($reflections as $reflection) {
+            if ($unique === true) {
+                $result = $this->annotationManager->getAnnotation(
+                    $reflection->getDocComment(),
+                    $annotation
+                );
+                if ($result !== null) {
+                    $annotations[$reflection->getName()] = $result;
+                }
+            } else {
```

Comment:
Codacy found an issue: [The method getAnnotations uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/pierregaimard/framework3/pullRequest?prid=6283010)